tests/unit_tests/test_cli.py

```diff
@@ -0,0 +1,62 @@
+import os
+from unittest.mock import patch
+
+import pytest
+from click.testing import CliRunner, Result
+
+from dodal import __version__
+from dodal.cli import main
+
+# Test with an example beamline, device instantiation is already tested
+# in beamline unit tests
+EXAMPLE_BEAMLINE = "i22"
+
+
+@pytest.fixture
+def runner():
+    return CliRunner()
+
+
+def test_cli_version(runner: CliRunner):
+    result = runner.invoke(
+        main,
+        ["--version"],
+        catch_exceptions=False,
+    )
+
+    assert result.stdout == f"{__version__}\n"
+
+
+def test_cli_sets_beamline_environment_variable(runner: CliRunner):
+    with patch.dict(os.environ, clear=True):
+        _mock_connect(EXAMPLE_BEAMLINE, runner=runner)
+        assert os.environ["BEAMLINE"] == EXAMPLE_BEAMLINE
+
+
+# We need to mock the environment because the CLI edits it and this could break other
+# tests
```

Comment:
Setting `os.environ["BEAMLINE"]` caused unintended side effects, will make an issue to find a better way

Reply:
Done in #569